feat(parser): Add `ClientSideToggleMenuItem`

jonz94 opened the pull request:
Fixes #595 
Fixes #727 

Example RawNode data 1:

```json
{
  "clientSideToggleMenuItemRenderer": {
    "defaultText": {
      "runs": [
        {
          "text": "Timestamps"
        }
      ]
    },
    "defaultIcon": {
      "iconType": "ACCESS_TIME"
    },
    "toggledText": {
      "runs": [
        {
          "text": "Timestamps"
        }
      ]
    },
    "toggledIcon": {
      "iconType": "ACCESS_TIME"
    },
    "menuItemIdentifier": "timestamp_toggle",
    "command": {
      "clickTrackingParams": "CAYQl98BIhMI7bKs8KG9iQMVWaPYBR2lDhra",
      "toggleLiveChatTimestampsEndpoint": {
        "hack": true
      }
    }
  }
}
```

Example RawNode data 2:

```json
{
  "clientSideToggleMenuItemRenderer": {
    "defaultText": {
      "runs": [
        {
          "text": "Reactions"
        }
      ]
    },
    "defaultIcon": {
      "iconType": "HEART_CIRCLE"
    },
    "toggledText": {
      "runs": [
        {
          "text": "Reactions"
        }
      ]
    },
    "toggledIcon": {
      "iconType": "HEART_CIRCLE"
    },
    "isToggled": true,
    "menuItemIdentifier": "reactions_toggle",
    "command": {
      "clickTrackingParams": "CBIQjcsMGDsiEwin892z6KCKAxXtAXsHHdGyB14=",
      "toggleLiveReactionsMuteCommand": {
        "hack": true
      }
    },
    "loggingDirectives": {
      "trackingParams": "CBIQjcsMGDsiEwin892z6KCKAxXtAXsHHdGyB14=",
      "visibility": {
        "types": "12"
      },
      "enableDisplayloggerExperiment": true
    }
  }
}
```

---

I notice that the `command` field is not a conventional `NavigationEndpoint`

```json
"command": {
  "clickTrackingParams": "CAYQl98BIhMI7bKs8KG9iQMVWaPYBR2lDhra",
  "toggleLiveChatTimestampsEndpoint": {
    "hack": true
  }
}
```

I'm not sure how to properly parse this.

Or maybe, should we just ignore the `command` field?
Since it does not interact with any backend endpoint.

LuanRT commented:
It's ok to parse the command. It's supposed to trigger a client side action, and people building YouTube clients that follow YouTube's SDUI rendering approach might find it useful.

jonz94 commented:
> It's ok to parse the command. It's supposed to trigger a client side action, and people building YouTube clients that follow YouTube's SDUI rendering approach might find it useful.

Thanks for the clarification! I've updated the parser to parse the `command` data.

---

I also found another occurrence of the `ClientSideToggleMenuItem`

```json
{
  "clientSideToggleMenuItemRenderer": {
    "defaultText": {
      "runs": [
        {
          "text": "Reactions"
        }
      ]
    },
    "defaultIcon": {
      "iconType": "HEART_CIRCLE"
    },
    "toggledText": {
      "runs": [
        {
          "text": "Reactions"
        }
      ]
    },
    "toggledIcon": {
      "iconType": "HEART_CIRCLE"
    },
    "isToggled": true,
    "menuItemIdentifier": "reactions_toggle",
    "command": {
      "clickTrackingParams": "CBIQjcsMGDsiEwin892z6KCKAxXtAXsHHdGyB14=",
      "toggleLiveReactionsMuteCommand": {
        "hack": true
      }
    },
    "loggingDirectives": {
      "trackingParams": "CBIQjcsMGDsiEwin892z6KCKAxXtAXsHHdGyB14=",
      "visibility": {
        "types": "12"
      },
      "enableDisplayloggerExperiment": true
    }
  }
}
```

which includes extra fields not present in the original implementation, so I updated the parser to account for it.
